mempool/reactor_test.go

```diff
@@ -685,3 +685,184 @@ func TestMempoolVectors(t *testing.T) {
 		require.Equal(t, tc.expBytes, hex.EncodeToString(bz), tc.testName)
 	}
 }
+
+// Verify that counting of duplicates and first time transactions work
+// The test sends transactions from node2 to node1 twice.
+// The second time they will get rejected.
+func TestDOGTransactionCount(t *testing.T) {
+	config := cfg.TestConfig()
+	config.Mempool.DOGProtocolEnabled = true
+
+	// Put the interval to a higher value to make sure the values don't get reset
+	config.Mempool.DOGAdjustInterval = 15 * time.Second
+	reactors, _ := makeAndConnectReactors(config, 2, nil)
+
+	// create random transactions
+	txs := NewRandomTxs(numTxs, 20)
+	secondNodeID := reactors[1].Switch.NodeInfo().ID()
+	secondNode := reactors[0].Switch.Peers().Get(secondNodeID)
+
+	txUnique := make(map[string][]byte, len(txs))
```

Comment:
Is `txUnique` really needed? Can `txs` have duplicate elements?

Reply:
there is a function that can be used for this, I edited the code 6ea2193f2